src/Aspire.Dashboard/Model/ResourceViewModel.cs

```diff
@@ -252,3 +252,51 @@ public bool MatchesFilter(string filter) =>
         Source?.Contains(filter, StringComparison.CurrentCultureIgnoreCase) == true ||
         Target?.Contains(filter, StringComparison.CurrentCultureIgnoreCase) == true;
 }
+
+public sealed record class HealthReportViewModel(string Name, HealthStatus HealthStatus, string? Description, string? Exception) : IPropertyGridItem
+{
+    /// <summary>
+    /// A single string that contains all information about this health report,
+    /// for copying and visualizing.
+    /// </summary>
+    private readonly string _compositeValue = GetCompositeValue(HealthStatus, Description, Exception);
+
+    string? IPropertyGridItem.Name => Name;
+
+    string? IPropertyGridItem.Value => HealthStatus.ToString();
+
+    string? IPropertyGridItem.ValueToCopy => _compositeValue;
+
+    string? IPropertyGridItem.ValueToVisualize => _compositeValue;
+
+    private static string GetCompositeValue(HealthStatus HealthStatus, string? Description, string? Exception)
+    {
+        StringBuilder builder = new();
+
+        builder.Append(HealthStatus.ToString());
+
+        if (!string.IsNullOrWhiteSpace(Description))
+        {
+            builder.AppendLine();
+            builder.AppendLine();
+            builder.Append(Description);
+        }
+
+        if (!string.IsNullOrWhiteSpace(Exception))
+        {
+            builder.AppendLine();
+            builder.AppendLine();
+            builder.Append(Exception);
+        }
+
+        return builder.ToString();
+    }
+
+    public bool MatchesFilter(string filter)
+    {
+        return
+            Name?.Contains(filter, StringComparison.CurrentCultureIgnoreCase) == true ||
+            Description?.Contains(filter, StringComparison.CurrentCultureIgnoreCase) == true ||
+            Exception?.Contains(filter, StringComparison.CurrentCultureIgnoreCase) == true;
```

Comment:
I think the problem with including exception here is it isn't visible on the page. While searching you don't see text it being highlighted. It could be confusing why the health check is still visible. IMO remove it.

Also, add a check on the humanized result of the healthstatus enum (see earlier comment about it being filtered out). Maybe `HealthStatus.Humanize()` should be cached somewhere. Don't want to repeatedly process it in a Where statement. I don't know how expensive it is.

Reply:
Looking at the implementation, it's moderately expensive.